### PR TITLE
Add kmod tests

### PR DESCRIPTION
**Pull request by BillAnastasiadis**
Add tests for kmod. Tests common commands: modinfo, lsmod, modprobe, rmmod, depmod. Tests the commands and their output for common words that should appear independently of the individual modules loaded in each system.

-Related issue: https://progress.opensuse.org/issues/55778
-Verification runs:
http://10.161.229.247/tests/47
http://10.161.229.247/tests/48
http://10.161.229.247/tests/49
http://10.161.229.247/tests/50
http://10.161.229.247/tests/51
http://10.161.229.247/tests/52
http://10.161.229.247/tests/56

### CODE REVIEW

File: tests/console/kmod.pm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    assert_script_run('grep -o -m 1 \'^license:.*GPL\' <<< "$OUT"');
+    assert_script_run('grep -o -m 1 \'^author:.*Qumranet\' <<< "$OUT"');
+    assert_script_run('grep -o -m 1 \'^parm:.*ignore_msrs:bool\' <<< "$OUT"');
+    #test 2
+    #test lsmod command
+    #test that the output has the expected correct format
+    assert_script_run('lsmod | grep \'^Module.*Size.*Used by$\'');
+    #test 3
+    #test modprobe -v arc4 command
+    script_run('rmmod arc4');    #make sure the module is not loaded
```
Comment (dzedro) [inline tests/console/kmod.pm]: If you really want to make sure use `-f` ?
On line 59 you use `assert_script_run`, any reason to use one or another way to run it ?

**Comment by pdostal** [comment]
Hello Bill, in my opinion the summary is too long. I would keep only what the test does and then people can go and see how it is done in the body of the test. Also maybe remove comments like `#test 2` and just keep one empty line between each scenario.

**Comment by BillAnastasiadis** [comment]
> Hello Bill, in my opinion the summary is too long. I would keep only what the test does and then people can go and see how it is done in the body of the test. Also maybe remove comments like `#test 2` and just keep one empty line between each scenario.

Hello Pavel! Thanks for your input. I had it smaller but Vit told me to make it detailed, in a way that even someone who can't read code can know exactly what is the test doing. - I will ask him again.

**Comment by okurz** [comment]
Hi @BillAnastasiadis I know that many many test modules have already received a quite extensive summary but I am afraid this is a quite dangerous path. Coding best practices tell that eventually someone will update the code and not update comments so they run out of sync. Comments should not describe what the code does because this would be redundant. At best it can summarize or describe the reasoning.

File: tests/console/kmod.pm (hunk 1 above)
Comment (BillAnastasiadis) [reply tests/console/kmod.pm]: Hi Josef, thanks for the input!
In line 55 I use script_run since I don't care about the exit code, I just wanted to make sure it's not loaded and rmmod it if it was, so I can test modprobe right after. 

Then, in line 59 I am actually testing rmmod so I care about whether it was successful or not (I know it is loaded because I used modprobe right before). That's why I assert this time. Then I run modprobe one last time to make sure that rmmod did the work.

**Comment by pdostal** [comment]
> > Hello Bill, in my opinion the summary is too long. I would keep only what the test does and then people can go and see how it is done in the body of the test. Also maybe remove comments like `#test 2` and just keep one empty line between each scenario.
> 
> Hello Pavel! Thanks for your input. I had it smaller but Vit told me to make it detailed, in a way that even someone who can't read code can know exactly what is the test doing. - I will ask him again.

@BillAnastasiadis @okurz I would suggest something like:
```
# Summary: Test common commands and their output for common words that
# should appear independently of the individual modules loaded in
# each system:
# * modinfo: 'filename ... .ko', 'license ... GPL', 'author ... Qumranet'
# * lsmod: 'Module', 'Size', 'Used by'
# * modprobe: We make sure arc4 module is not active and then we activate it
# * rmmod: We check the exit status and then we enable the disabled module again
# * depmod: 'lib/modules', '.ko', 'kernel'.
```

**Comment by BillAnastasiadis** [comment]
> would suggest something like:
> 
> ```
> # Summary: Test common commands and their output for common words that
> # should appear independently of the individual modules loaded in
> # each system:
> # * modinfo: 'filename ... .ko', 'license ... GPL', 'author ... Qumranet'
> # * lsmod: 'Module', 'Size', 'Used by'
> # * modprobe: We make sure arc4 module is not active and then we activate it
> # * rmmod: We check the exit status and then we enable the disabled module again
> # * depmod: 'lib/modules', '.ko', 'kernel'.
> ```

I agree with this, I am going to change it now. Thanks for your suggestion Pavel!

**Comment by BillAnastasiadis** [comment]
> Hi @BillAnastasiadis I know that many many test modules have already received a quite extensive summary but I am afraid this is a quite dangerous path. Coding best practices tell that eventually someone will update the code and not update comments so they run out of sync. Comments should not describe what the code does because this would be redundant. At best it can summarize or describe the reasoning.

I agree. I am going with Pavel's suggestion. Thanks for the input!

File: tests/console/kmod.pm (hunk 1 above)
Comment (dzedro) [reply tests/console/kmod.pm]: good, thank you

**Comment by dzedro** [comment]
https://openqa.opensuse.org/tests/1028250

**Comment by ggardet** [comment]
It fails on aarch64 as kvm module is built-in, not a module: https://openqa.opensuse.org/tests/1028391#step/kmod/4

**Comment by SergioAtSUSE** [comment]
Test failing on JeOS:
- https://progress.opensuse.org/issues/56699
- https://openqa.opensuse.org/tests/1028391#step/kmod/4

@BillAnastasiadis, could you take a look?

**Comment by ggardet** [comment]
> Test failing on JeOS:
> 
> * https://progress.opensuse.org/issues/56699
> * https://openqa.opensuse.org/tests/1028391#step/kmod/4
> 
> @BillAnastasiadis, could you take a look?

There is a PR already: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8417
